drivers/serial/uart_nrf5.c

```diff
@@ -221,29 +162,34 @@ static int uart_nrf5_init(struct device *dev)
 				  CONFIG_UART_NRF5_GPIO_CTS_PIN,
 				  (GPIO_DIR_IN));
 
-	uart->PSELRTS = CONFIG_UART_NRF5_GPIO_RTS_PIN;
-	uart->PSELCTS = CONFIG_UART_NRF5_GPIO_CTS_PIN;
-	uart->CONFIG = (UART_CONFIG_HWFC_Enabled << UART_CONFIG_HWFC_Pos);
-
+	nrf_uart_hwfc_pins_set(uart,
+			       CONFIG_UART_NRF5_GPIO_RTS_PIN,
+			       CONFIG_UART_NRF5_GPIO_CTS_PIN);
+	nrf_uart_configure(uart,
+			   NRF_UART_PARITY_EXCLUDED,
+			   NRF_UART_HWFC_ENABLED);
+#else
+	nrf_uart_configure(uart,
+			   NRF_UART_PARITY_EXCLUDED,
+			   NRF_UART_HWFC_DISABLED);
 #endif /* CONFIG_UART_NRF5_FLOW_CONTROL */
 
 	DEV_DATA(dev)->baud_rate = CONFIG_UART_NRF5_BAUD_RATE;
 
 	/* Set baud rate */
-	err = baudrate_set(dev, DEV_DATA(dev)->baud_rate,
-		     DEV_CFG(dev)->sys_clk_freq);
+	err = baudrate_set(dev, DEV_DATA(dev)->baud_rate);
 	if (err) {
 		return err;
 	}
 
 	/* Enable receiver and transmitter */
-	uart->ENABLE = (UART_ENABLE_ENABLE_Enabled << UART_ENABLE_ENABLE_Pos);
+	nrf_uart_enable(uart);
 
-	uart->EVENTS_TXDRDY = 0;
-	uart->EVENTS_RXDRDY = 0;
+	nrf_uart_event_clear(uart, NRF_UART_EVENT_TXDRDY);
+	nrf_uart_event_clear(uart, NRF_UART_EVENT_RXDRDY);
 
-	uart->TASKS_STARTTX = 1;
-	uart->TASKS_STARTRX = 1;
+	nrf_uart_task_trigger(uart, NRF_UART_TASK_STARTTX);
+	nrf_uart_task_trigger(uart, NRF_UART_TASK_STARTRX);
 
 	dev->driver_api = &uart_nrf5_driver_api;
```

Comment:
This should be done by the `DEVICE_AND_API_INIT` macro which should be used instead of `DEVICE_INIT` at the end of this file.